pontoon/base/models.py

```diff
@@ -2868,22 +2868,25 @@ def update_latest_translation(self):
         resource = self.entity.resource
         project = resource.project
         locale = self.locale
-        translatedresource = TranslatedResource.objects.get(resource=resource, locale=locale)
 
-        instances = [translatedresource, project]
+        to_update = [
+            (TranslatedResource, Q(Q(resource=resource) & Q(locale=locale))),
+            (ProjectLocale, Q(Q(project=project) & Q(locale=locale))),
+            (Project, Q(pk=project.pk)),
+        ]
 
         if not project.system_project:
-            instances.append(locale)
-
-        project_locale = utils.get_object_or_none(ProjectLocale, project=project, locale=locale)
-        if project_locale:
-            instances.append(project_locale)
+            to_update.append((Locale, Q(pk=locale.pk)))
 
-        for instance in instances:
-            latest = instance.latest_translation
-            if latest is None or self.latest_activity['date'] > latest.latest_activity['date']:
-                instance.latest_translation = self
-                instance.save(update_fields=['latest_translation'])
+        for model, query in to_update:
+            model.objects.filter(
+                Q(
+                    query & Q(
+                        Q(latest_translation=None)
+                        | Q(latest_translation__date__lt=self.latest_activity['date'])
```

Comment:
Nit: Please move that `|` operator to the end of the previous line, so that `Q()`'s will align.

Reply:
Done.